types/eslint/index.d.ts

```diff
@@ -1083,9 +1084,41 @@ export namespace Linter {
     }
 
     /**
-     * Parser options.
+     * Parser options for flat config.
+     *
+     * This properties are used to pass options directly to parsers. These options are always parser-specific,
+     * so you’ll need to check the documentation of the parser you’re using for available options.
+     *
+     * @see [Specifying Parser Options](https://eslint.org/docs/latest/use/configure/language-options#specifying-parser-options)
+     * @see [Configure Parser Options](https://eslint.org/docs/latest/use/configure/parser#configure-parser-options)
+     */
+    interface FlatParserOptions {
```

Comment:
memo: The parser options for legacy config have `ecmaVersion` and `sourceType` properties, but the parser options for flat config do not have them.